### PR TITLE
WIP: Add initial webconfig from fish

### PR DESCRIPTION
**Pull request by Emiller88**
Start on #1798 during scipy19.

### CODE REVIEW

**Comment by scopatz** [comment]
Thanks for starting on this @Emiller88! Has there been any more progress?

**Comment by Emiller88** [comment]
@scopatz no not yet, I've been busy with other things, if you'd like to take
over feel free!

**Comment by scopatz** [comment]
OK thanks! I think this is open for folks to help out with then!

File: web_config/webconfig.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/python
```
Comment (jb-leger) [inline web_config/webconfig.py]: The `/usr/bin/env python3` sheband should be used to allow user changing python exec from its env.

File: web_config/webconfig.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+
+try:  # Python2
```
Comment (jb-leger) [inline web_config/webconfig.py]: xonsh doesn't not support python-legacy (a.k.a. python2), you can assume python3 is used.

File: web_config/webconfig.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    import SimpleHTTPServer
+except ImportError:  # Python3
+    import http.server as SimpleHTTPServer
+try:  # Python2
```
Comment (jb-leger) [inline web_config/webconfig.py]: same comment about python2/3.

File: web_config/webconfig.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    import SocketServer
+except ImportError:  # Python3
+    import socketserver as SocketServer
+import webbrowser
+import subprocess
+import re, json, socket, os, sys, cgi, select
```
Comment (jb-leger) [inline web_config/webconfig.py]: this import should be splitted in many imports.

**Comment by jb-leger** [comment]
Maybe we have a licensing issue with this PR. This code is adapted from fish [webconfig](https://github.com/fish-shell/fish-shell/blob/master/share/tools/web_config/webconfig.py), and I didn't find that the webconfig is under a different license than the code code of fish, *i.e.* GPLv2.

It could be not possible to merge this code in xonsh for this point, I think we have to consider this point.

**Comment by scopatz** [comment]
As per licensing, I think it depends. Everything that is here now could be written to be sufficently different

**Comment by Emiller88** [comment]
@jb-leger Thanks for the feedback! 

For reference here is [fish's webconfig](https://github.com/fish-shell/fish-shell/tree/master/share/tools/web_config)

**Pull request merged**